docs/orleans/grains/grain-identity.md

```diff
@@ -1,33 +1,71 @@
 ---
 title: Grain identity
 description: Learn about grain identities in .NET Orleans.
-ms.date: 03/16/2022
+ms.date: 12/12/2023
 ---
 
 # Grain identity
 
-In object-oriented environments, the identity of an object is hard to distinguish from a reference to it. Thus, when an object is created using new, the reference you get back represents all aspects of its identity except those that map the object to some external entity that it represents.
+Grains in Orleans each have a single, unique, user-defined identifier which consists of two parts:
 
-In distributed systems, object references cannot represent instance identity, since references are typically limited to a single address space. That is certainly the case for .NET references. Furthermore, a grain must have an identity regardless of whether it is active, so that we can activate it on demand. Therefore grains have a primary key. The primary key can be a <xref:System.Guid>, a <xref:System.Int64>, or a <xref:System.String>.
+1. The grain _type_ name, which uniquely identifies the grain class.
+2. The grain _key_, which uniquely identifies a logical instance of that grain class.
 
+The grain type and key are both represented as human-readable strings in Orleans and, by convention, the grain identity is written with the grain type and key separated by a `/` character. For example, `shoppingcart/bob65` represents the grain type named `shoppingcart` with a key `bob65`.
+
+It is not common to construct grain identities directly. Instead, it is more common to create [grain references](./grain-references.md) using <xref:Orleans.IGrainFactory?displayProperty=nameWithType>.
```

Comment:
```suggestion
It's not common to construct grain identities directly. Instead, it's more common to create [grain references](./grain-references.md) using <xref:Orleans.IGrainFactory?displayProperty=nameWithType>.
```

Reply:
I need to get used to using contractions liberally